src/wallet/wallet.cpp

```diff
@@ -1514,13 +1514,46 @@ int64_t CalculateMaximumSignedTxSize(const CTransaction &tx, const CWallet *wall
     return GetVirtualTransactionSize(txNew);
 }
 
+size_t CalculateNestedKeyhashInputSize(bool use_max_sig)
```

Comment:
Is this function just returning one of two integers that could be precomputed? If so, this seems a bit excessive.

You could leave the function as a comment to show it was computed.

Reply:
Longer-term I was hoping to have utility functions for various sorts of standard input types.

> You could leave the function as a comment to show it was computed.

Sorry I'm not quite parsing what you mean here.

Reply:
A related suggestion: you could directly return the expected literal in this method, then move this implementation code to a test which asserts that the production code returns the same value produced by the current implementation here. Would be good then to reference the test here in a comment to explain the literal result. That way the association between complete implementation and literal optimization is self-enforcing going forward.

Reply:
That's actually quite reasonable, thank you.